test/test_prior.py

```diff
@@ -1,23 +1,115 @@
+        p = UniformPrior(xmin, xmax, ["x"])
+        # Check that all the samples are finite
+        samples = p.sample(jax.random.PRNGKey(0), 10000)
+        assert jnp.all(jnp.isfinite(samples['x']))
+        # Check that the log_prob is correct in the support
+        samples = trace_prior_parent(p, [])[0].sample(jax.random.PRNGKey(0), 10000)
+        log_prob = jax.vmap(p.log_prob)(samples)
+        assert jnp.allclose(log_prob, jnp.log(1.0 / (xmax - xmin)))
 
-class TestPriorOperations:
-    def test_combine(self):
-        raise NotImplementedError
+    def test_sine(self):
+        p = SinePrior(["x"])
+        # Check that all the samples are finite
+        samples = p.sample(jax.random.PRNGKey(0), 10000)
+        assert jnp.all(jnp.isfinite(samples['x']))
+        # Check that the log_prob is finite
+        samples = trace_prior_parent(p, [])[0].sample(jax.random.PRNGKey(0), 10000)
+        log_prob = jax.vmap(p.log_prob)(samples)
+        assert jnp.all(jnp.isfinite(log_prob))
+        # Check that the log_prob is correct in the support
+        x = trace_prior_parent(p, [])[0].add_name(jnp.linspace(-10.0, 10.0, 1000)[None])
+        y = jax.vmap(p.base_prior.base_prior.transform)(x)
+        y = jax.vmap(p.base_prior.transform)(y)
+        y = jax.vmap(p.transform)(y)
+        assert jnp.allclose(jax.vmap(p.log_prob)(x), jnp.log(jnp.sin(y['x'])/2.0))
+        
+    def test_cosine(self):
+        p = CosinePrior(["x"])
+        # Check that all the samples are finite
+        samples = p.sample(jax.random.PRNGKey(0), 10000)
+        assert jnp.all(jnp.isfinite(samples['x']))
+        # Check that the log_prob is finite
+        samples = trace_prior_parent(p, [])[0].sample(jax.random.PRNGKey(0), 10000)
+        log_prob = jax.vmap(p.log_prob)(samples)
+        assert jnp.all(jnp.isfinite(log_prob))
+        # Check that the log_prob is correct in the support
+        x = trace_prior_parent(p, [])[0].add_name(jnp.linspace(-10.0, 10.0, 1000)[None])
+        y = jax.vmap(p.base_prior.transform)(x)
+        y = jax.vmap(p.transform)(y)
+        assert jnp.allclose(jax.vmap(p.log_prob)(x), jnp.log(jnp.cos(y['x'])/2.0))
 
-    def test_sequence(self):
-        raise NotImplementedError
+    def test_uniform_sphere(self):
+        p = UniformSpherePrior(["x"])
+        # Check that all the samples are finite
+        samples = p.sample(jax.random.PRNGKey(0), 10000)
+        assert jnp.all(jnp.isfinite(samples['x_mag']))
+        assert jnp.all(jnp.isfinite(samples['x_theta']))
+        assert jnp.all(jnp.isfinite(samples['x_phi']))
+        # Check that the log_prob is finite
+        samples = {}
+        for i in range(3):
+            samples.update(trace_prior_parent(p, [])[i].sample(jax.random.PRNGKey(0), 10000))
+        log_prob = jax.vmap(p.log_prob)(samples)
+        assert jnp.all(jnp.isfinite(log_prob))
+    
+    def test_power_law(self):
+        from bilby.core.prior.analytical import PowerLaw
```

Comment:
Bilby is not a dependence of jim, and I don't think we should add it even in the test. Please remove this